### ISSUE TITLE
I need to be able to delete a problem set that I've created

### ISSUE THREAD

**Issue by benetechMatt** [benetech/MathShare#844]
Add this as an option to the more options button in the dashboard.  Can delete problem sets the user created or copied

Aha! Link: https://diagramlabs.aha.io/features/B-87

**Comment by johnhbenetech** [comment]
Can you clarify what exactly this does? I see two options:

1. The problem set is simply disassociated with the user. It's URL still exists as well as any student responses attached to it. It will simply no longer appear on the dashboard.

2. The problem set is legitimately removed from the database.


If we are talking about option 2, please enumerate exactly what should happen to the URL, any previous versions of the problem set, and any attached student responses.

**Comment by benetechMatt** [comment]
We could go with 1 and call it "archive". @alexrcabral what are your thoughts on this? an archived problem implies that it's not completely lost and still available some how.  If a student clicks an archived set it could still be available but have a warning that the set has now been archived. Users would access archived sets by selecting a new link at the bottom of their sets that says "view archived problem sets"

**Comment by johnhbenetech** [comment]
Pre-pend 'Archived' to any share/solution set title associated with an archived set.

Add gray (rgb(90, 98, 104)) 'Archived Sets' button to dashboard on lower right below Recent section. This will bring you to a screen that looks like the dashboard but listing only archived sets. These sets should have an option under the three dot menu to 'Reactivate' which will bring it back to the main dashboard within the 'recent' section. The Archived dashboard also needs a back button in the upper left corner 'Back to Dashboard'.

Please check in front end commits separately so @jscholes can review more easily.

**Comment by bognasoldev89** [comment]
Verified on QA env

**Verified cases**

1. Archiving sets from the homepage
2. Archiving sets from the set page
3. Accessing archived sets page

Ad. 1
I was able to archive Problem set from the homepage

![ms 844 1](https://user-images.githubusercontent.com/51324797/80363156-18b81800-8884-11ea-863e-61b96d25f1a2.gif)

Ad. 2
I was not able to archive Problem set from its page.

![ms 844 2](https://user-images.githubusercontent.com/51324797/80362477-02f62300-8883-11ea-8b84-9829d1fc7d8f.gif)

@johnhbenetech @benetechMatt please confirm that Archive option should or should not be available from this level.
Moreover, I created this bug regarding Options for Problems set from its page here #1151 

Ad.3
I was not able to access Archived sets page, it goes blank after clicking "Archived sets" button. Moreover, after clicking the browsers back button, the homepage is also blank. User has to refresh the page to access the app.

![ms 844 3](https://user-images.githubusercontent.com/51324797/80362494-07224080-8883-11ea-9596-bc8371f12419.gif)

**Additional issues:**

Ad. 4
From @johnhbenetech comment, I understood that "Archive button" should be below Recent sets section, not above. I put a red rectangle in a place where I imagined this button:

![ms 844 4](https://user-images.githubusercontent.com/51324797/80363110-06d67500-8884-11ea-8710-02a91938d7de.png)

Ad. 5
@johnhbenetech @benetechMatt can you please confirm for which users this feature should be available? At this moment only Teacher can archive problem set.

**Comment by rupeshparab** [comment]
@bognasoldev89 
I have pushed a fix for the 3rd issue which you have mentioned

Will wait on input for the other issues

**Comment by johnhbenetech** [comment]
@alexrcabral do you think archive set should also be available under the three dot menu on the single problem set edit view?

**Comment by abaranowska1** [comment]
@rupeshparab  @bognasoldev89 
Confirmed that the 3rd issue was fixed. Now user can Archive and Restore Problem Set. Page with Archived Sets works and opens. 

We will wait for the rest fixes.

**Comment by alexrcabral** [comment]
@johnhbenetech Archive set should not be available from the single problem set edit view, since the effect could be difficult for users to follow. 

And to answer @bognasoldev89's question, any logged in user should be able to archive a problem set (Students, Teachers, and Others).

**Comment by rupeshparab** [comment]
@alexrcabral 
Currently students can only see their solutions to Problem Sets in the Recent section, because of this archive option is not shown to them as only Problem Sets can be archived. When we make changes to recent set functionality such that solutions and problem sets both are shown to the users in recent set, then students will also be able to Archive Problem sets created by them

Also, can you please confirm if the current placement of Archived sets looks fine or it should be move to the red mark:
![image](https://user-images.githubusercontent.com/6897505/80780381-61c1e200-8b8c-11ea-8dac-22085af0ed57.png)

**Comment by johnhbenetech** [comment]
@alexrcabral Confirm button placement is correct.

Also @alexrcabral - note here there are two things going on:

* #1175 will display problem sets created by students as well - we can then add the button to archive

* Still, there isn't any functionality right now to archive 'Solution Sets' (only problem sets). So if you want students to eventually have that ability that should be broke out in a new task.

**Comment by alexrcabral** [comment]
Button placement should be moved to red mark as recommended by @bognasoldev89 

And thanks for the clarification @rupeshparab, I was mistaken in thinking that the created sets and solvable sets were treated more similarly than they are. I've added #1178 to address this.

**Comment by jscholes** [comment]
@rupeshparab Could you please explain where the control was, and where you have moved it in https://github.com/benetech/MathShare/pull/1180?  @bognasoldev89 Your screenshots have no alt text, so as a screen reader user I have no idea what they show.  Please ensure that all screenshots and other images posted here have descriptive alternative text, otherwise it is impossible for us to collaborate on an equal footing.

**Comment by rupeshparab** [comment]
@jscholes 
The archived set button was on the right end of the 'Recent Sets' header, I have moved it below the recent sets list items on right side

**Comment by bognasoldev89** [comment]
**Tested with success**

Verified on QA env

**Verified cases**
- Archiving problem set
- Restoring problem set

**Summary**
This feature works fine now. I didn't spot the issues I found earlier.  3 dot menu on problem set page doesn't allow the archiving problem set.
At this moment only a teacher can archive problem sets, but as I understood, students and others will gain this feature in the scope of other tickets cc: @johnhbenetech @alexrcabral 
"Archived sets" button is moved below the recent sets section.

**Attachments**
- Correct placement of "Archived sets" button (below the recent sets section to the right)
![ms 844 archived sets button](https://user-images.githubusercontent.com/51324797/81794784-99754600-950b-11ea-8452-0cb50403830b.png)

- Correct archiving and restoring problem set
![844 archiving and restoring](https://user-images.githubusercontent.com/51324797/81794883-bd388c00-950b-11ea-8c99-16efac29da9f.gif)

- 3 dot menu on problem set page, as a confirmation that the user is not allowed to archive set from this place
![844 set 3 dot menu](https://user-images.githubusercontent.com/51324797/81794979-dc371e00-950b-11ea-8c76-15c3e2bebd97.png)
